src/uu/ls/src/ls.rs

```diff
@@ -65,107 +69,291 @@ lazy_static! {
     static ref END_CODE: &'static str = COLOR_MAP.get("ec").unwrap_or(&"");
 }
 
+pub mod options {
+    pub static ONELINE: &str = "1";
+    pub static ALL: &str = "all";
+    pub static ALMOST_ALL: &str = "almost-all";
+    pub static IGNORE_BACKUPS: &str = "ignore-backups";
+    pub static DIRECTORY: &str = "directory";
+    pub static CLASSIFY: &str = "classify";
+    pub static HUMAN_READABLE: &str = "human-readable";
+    pub static INODE: &str = "inode";
+    pub static DEREFERENCE: &str = "dereference";
+    pub static LONG: &str = "long";
+    pub static NUMERIC_UID_GID: &str = "numeric-uid-gid";
+    pub static REVERSE: &str = "reverse";
+    pub static RECURSIVE: &str = "recursive";
+    pub static SORT_SIZE: &str = "S";
+    pub static SORT_TIME: &str = "t";
+    pub static SORT_NONE: &str = "U";
+    pub static SORT_CTIME: &str = "c";
+    pub static COLOR: &str = "color";
+    pub static PATHS: &str = "paths";
+}
+
+#[derive(PartialEq, Eq)]
+enum DisplayOptions {
+    Columns,
+    Long,
+    OneLine,
+}
+
+enum Sort {
+    None,
+    Name,
+    Size,
+    Time,
+    CTime,
+}
+
+enum SizeFormats {
+    Bytes,
+    Binary, // Powers of 1024, --human-readable
+}
+
+#[derive(PartialEq, Eq)]
+enum Files {
+    All,
+    AlmostAll,
+    Normal,
+}
+
+struct Config {
+    display: DisplayOptions,
+    files: Files,
+    sort: Sort,
+    recursive: bool,
+    reverse: bool,
+    dereference: bool,
+    classify: bool,
+    ignore_backups: bool,
+    size_format: SizeFormats,
+    numeric_uid_gid: bool,
+    directory: bool,
+    #[cfg(unix)]
+    inode: bool,
+    #[cfg(unix)]
+    color: bool,
+}
+
+impl Config {
+    fn from(options: clap::ArgMatches) -> Config {
+        let display = if options.is_present(options::LONG) {
+            DisplayOptions::Long
+        } else if options.is_present(options::ONELINE) {
+            DisplayOptions::OneLine
+        } else {
+            DisplayOptions::Columns
+        };
+
+        let files = if options.is_present(options::ALL) {
+            Files::All
```

Comment:
As the options seem to be exclusive, maybe following this example?
https://github.com/uutils/coreutils/blob/517b2a6a2244637a5f5331565592873a734085c7/src/uu/chown/src/chown.rs#L30